.github/workflows/ci_v2.yml

```diff
@@ -0,0 +1,100 @@
+name: ci_v2
+
+on:
+  pull_request_target:
+    branches:
+      - develop
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request_target' && github.event.pull_request.number || github.ref_name }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request_target' }}
+    
+jobs:
+  authorize:
+    environment:
+      ${{ (github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository) &&
+      'external' || 'internal' }}
```

Comment:
This `external` vs `internal` environment would differentiate between regular PRs and PRs incoming from forks and we can require approval and we can specify that `external` ones need approval before running.